wurst/data/LinkedList.wurst

```diff
@@ -327,6 +329,43 @@ public interface LLItrClosure<T>
 public interface LinkedListUpdater<T>
 	function update(T t) returns T
 
+/*  String processing extra functions 
+	The only reason this is here is that it can't be in String.wurst due to dependency problems */
+
+/** Splits the string by another string 
+	"this..is..an..example" split by the string ".." splits into a list containing four strings ["this", "is", "an", "example"] */
+public function string.split(string splitBy) returns LinkedList<string>
+	let list = new LinkedList<string>
+	let offset = splitBy.length()
+	var str = this
+	while true
+		let splitIndex = str.indexOf(splitBy)
+		if splitIndex == -1
+			list.add(str)
+			break
+		let first  = str.substring(0, splitIndex)
+		let second = str.substring(splitIndex + offset, str.length())
+		list.add(first)
+		str = second
+	return list
+
+/** Joins elements from a string list into one string using a separator */
+public function LinkedList<string>.joinBy(string separator) returns string
+	var joined = ""
+	let iter = this.iterator()
+	for str from iter
+		if iter.hasNext()
+			joined += str + separator
```

Comment:
I see why you want the iterator now.

May I suggest something more like

```
def list join iter:
    case list.len
        0 => ""
        1 => list.head
        otherwise => {
            var ret = list.head
            let iterator = list.iter
            while iterator.hasNext()
                ret += (iter + iterator.next())
            ret
        }
```